[6.3] qe_test_coverage BZ1406076

ldjebran opened the pull request:
cover : https://bugzilla.redhat.com/show_bug.cgi?id=1406076

```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_activationkey.py::ActivationKeyTestCase::test_positive_view_subscriptions_by_non_admin_user
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-12 11:35:39 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1378009', '1245334', '1217635', '1226425', '1147100', '1311113', '1199150', '1204686', '1221971', '1103157', '1230902', '1230865', '1214312', '1079482']

2017-09-12 11:35:39 - conftest - DEBUG - Collected 1 test cases


tests/foreman/cli/test_activationkey.py::ActivationKeyTestCase::test_positive_view_subscriptions_by_non_admin_user PASSED

================================================== 0 tests deselected ==================================================
============================================== 1 passed in 163.38 seconds ==============================================
```

abalakh commented on tests/foreman/cli/test_activationkey.py:
```diff
@@ -1452,3 +1455,115 @@ def test_positive_remove_user(self):
+            'organization-id': org['id'],
+            'id': activation_key['id'],
+        }, output_format='csv')
+        self.assertEqual(len(subscriptions), 1)
```
And what if empty string is returned? 🙂 
Please also add a check that actual subscription is present in output (probably by subscription name).

ldjebran replied:
format csv, this a table format I think, anyway if empty string is return the assertion will fail, will add to check that the subscription id is the same

abalakh replied:
```python
In [3]: subscriptions = ['']

In [4]: len(subscriptions)
Out[4]: 1
```
What do you mean? It will pass the check. Btw it's pretty common output - for some commands extra empty string is returned, so in case command returned nothing one empty string is present. Not sure about specific commands affected by the issue, but apparently there are some.

ldjebran replied:
any way this command ha no empty string the id checking is added that will prevent passing in such case, see updated code

ldjebran commented:
@abalakh comment addressed
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_activationkey.py::ActivationKeyTestCase::test_positive_view_subscriptions_by_non_admin_user
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-12 12:51:46 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_activationkey.py::ActivationKeyTestCase::test_positive_view_subscriptions_by_non_admin_user PASSED

============================================== 1 passed in 156.86 seconds ==============================================

```

ldjebran commented:
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_activationkey.py::ActivationKeyTestCase::test_positive_view_subscriptions_by_non_admin_user
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-12 17:48:38 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_activationkey.py::ActivationKeyTestCase::test_positive_view_subscriptions_by_non_admin_user PASSED

============================================== 1 passed in 175.34 seconds ==============================================
```

```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_with_non_admin_user
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-12 17:57:14 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_with_non_admin_user <- robottelo/decorators/__init__.py PASSED

============================================== 1 passed in 181.65 seconds ==============================================
```

ldjebran commented:
removed add_permissions_to_user , and refactored that test using it, to be be able to use a more general function add_role_permissions 
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_activationkey.py::ActivationKeyTestCase::test_positive_view_subscriptions_by_non_admin_user
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-12 18:34:14 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_activationkey.py::ActivationKeyTestCase::test_positive_view_subscriptions_by_non_admin_user PASSED

============================================== 1 passed in 168.85 seconds ==============================================
```

```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_with_non_admin_user
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 1 item 
2017-09-12 18:34:12 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_classparameters.py::SmartClassParametersTestCase::test_positive_list_with_non_admin_user <- robottelo/decorators/__init__.py PASSED

============================================== 1 passed in 187.34 seconds ==============================================
```
